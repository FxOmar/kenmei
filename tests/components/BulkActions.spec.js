import BulkActions from '@/components/BulkActions.vue';

describe('BulkActions.vue', () => {
  it('renders renders bulk actions button group', () => {
    const bulkActions = shallowMount(BulkActions);

    expect(bulkActions.html()).toMatchSnapshot();
  });

  describe('when pressing a button', () => {
    it('emits associated action', async () => {
      const bulkActions = shallowMount(BulkActions);

      await bulkActions.findAll('button').at(0).trigger('click');
      expect(bulkActions.emitted('delete')).toBeTruthy();

      await bulkActions.findAll('button').at(1).trigger('click');
      expect(bulkActions.emitted('edit')).toBeTruthy();

      await bulkActions.findAll('button').at(2).trigger('click');
      expect(bulkActions.emitted('read')).toBeTruthy();

      await bulkActions.findAll('button').at(3).trigger('click');
      expect(bulkActions.emitted('report')).toBeTruthy();
    });
  });
});
