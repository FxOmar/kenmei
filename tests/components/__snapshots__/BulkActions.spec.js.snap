// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkActions.vue renders renders bulk actions button group 1`] = `
<transition-stub enterclass="opacity-0 transform translate-y-8" leavetoclass="opacity-0 transform translate-y-8" enteractiveclass="slide-transition" leaveactiveclass="slide-transition"><span class="relative z-0 inline-flex shadow-sm"><button class="flex justify-center group rounded-l-md"><icontrash-stub class="-ml-1 mr-2 h-5 w-5 text-gray-400 group-hover_text-red-400"></icontrash-stub><span class="group-hover_text-red-400">Delete</span></button><button class="flex justify-center group -ml-px">
    <iconedit-stub class="-ml-1 mr-2 h-5 w-5 text-gray-400 group-hover_text-gray-800"></iconedit-stub><span class="group-hover_text-gray-800">Edit</span>
  </button><button class="flex justify-center group -ml-px">
    <iconcheckcircle-stub class="-ml-1 mr-2 h-5 w-5 text-gray-400 group-hover_text-green-400"></iconcheckcircle-stub><span class="group-hover_text-green-400">Read</span>
  </button><button class="flex justify-center group -ml-px rounded-r-md">
    <iconwarning-stub class="-ml-1 mr-2 h-5 w-5 text-gray-400 group-hover_text-yellow-400"></iconwarning-stub><span class="group-hover_text-yellow-400">Report</span>
  </button></span></transition-stub>
`;
